### PR TITLE
Herumi master

### PR DESCRIPTION
**Pull request by herumi**
update the latest version of herumi/mcl.

### CODE REVIEW

**Comment by enzoh** [comment]
Thanks for the alert @timohanke, will not be able to review until tomorrow

File: include/mcl/lagrange.hpp [hunk 1]
```diff
@@ -15,14 +15,19 @@ namespace mcl {
 template<class G, class F>
 void LagrangeInterpolation(bool *pb, G& out, const F *S, const G *vec, size_t k)
 {
+	if (k == 0) {
+		*pb = false;
+		return;
+	}
+	if (k == 1) {
+		out = vec[0];
```
Comment (enzoh) [inline include/mcl/lagrange.hpp:23]: Awesome thanks 👍

**Review by enzoh — APPROVED** [review]

**Comment by timohanke** [comment]
The line https://github.com/dfinity/mcl/blob/b3889f3bd981ab0b2449fd09822634a9c365236e/include/mcl/bn.h#L232
is wrong, or? It is for the old function `setLittleEndian` without `Mod`.

It says
> // mask buf with (1 << (bitLen(r) - 1)) - 1 if buf >= r

but the check "if buf >= r" never happens. The mask is always applied.

**Comment by herumi** [comment]
The comment is insufficient.
The old function setLittleEndian treats x as the following:
1) x &= (1 << bitLen(r)) - 1;
2) if (x >= r) x &= (1 << (bitLen(r) - 1)) - 1;

The reason for the two steps is to be able to accept x such that (1 << (bitLen(r) - 1)) <= x < p as is.

I will fix the comment later.

**Comment by herumi** [comment]
I fixed the comment.

**Comment by herumi** [comment]
The important patches are
* https://github.com/dfinity/mcl/pull/1/commits/54f2e8cc85d017945a47629d1986c29c291cf6eb ; for wasm
* https://github.com/dfinity/mcl/pull/1/commits/ed6ad685c2096769d5e7672dddad09b04d7eecde ; for wasm
* https://github.com/dfinity/mcl/pull/1/commits/f3f9be495c6a9c37340e7fe4879b945afc060c92 ; for performance
The other patches are not important.

**Pull request merged**